### PR TITLE
doc: reserved keywords update based on token.v

### PR DESCRIPTION
**Pull request by mobarski**
I've updated the reserved keywords list based on token.v

Each keyword was tested in the REPL with `keyword := 1` expression to be sure that it's in fact reserved.

```v
as
asm
assert
atomic
break
const
continue
defer
else
embed
enum
false
fn
for
go
goto
if
import
in
interface
is
lock
match
module
mut
none
or
pub
return
rlock
select
shared 
sizeof
static
struct
switch
true
type
typeof
union
unsafe
```

### CODE REVIEW

File: doc/docs.md [hunk 1]
```diff
@@ -2527,16 +2530,25 @@ import
...
 or
 pub
 return
+rlock
+select
+shared 
+sizeof
+static
 struct
+switch
```
Comment (Delta456) [inline doc/docs.md:2547]: `switch` is removed with match.

Comment (mobarski) [reply doc/docs.md:2547]: It's still reserved. I'll update this PR when your [PR](https://github.com/vlang/v/pull/6056) is merged.

Comment (mobarski) [reply doc/docs.md]: done

**Comment by ntrel** [comment]
I think just because something is a reserved word in the compiler doesn't make it a language keyword.

Some of the concurrency reserved words may be language keywords, but I think any unusual and dangerous concurrency keywords could be prefixed with 2 underscores (which are all reserved like in C).

`typeof` currently produces a string but should produce a type, and it could probably overload the `type` keyword instead, avoiding breakage when the change happens. (Then `typeof(expr).name` gives a string).

`sizeof` could be a type property, e.g. `int.sizeof` to save a keyword.

I don't know why `union` is a keyword, if supported I think it could be double underscore prefixed too.

**Comment by Delta456** [comment]
@ntrel I think `union` is a keyword due to C interpolation and some vlib code uses it.

**Comment by mobarski** [comment]
@ntrel [I agree](https://en.wikipedia.org/wiki/Reserved_word) but I didn't want to complicate things. Do you think we should split this list into keywords and reserved words? Union is a example of a reserved word which is not a keyword. It is a example of a keyword which is not a reserved word.

**Comment by ntrel** [comment]
I think any word which is a token in the compiler only because it is reserved in C, is not necessarily a reserved word in V unless you are using the C backend to generate C code. There's a difference in reserving a word in the V compiler implementation vs having it intentionally a reserved word in the language.

**Comment by medvednikov** [comment]
embed is not a keyword, it can be removed

I think sizeof and typeof should be considered as builtin fns

**Comment by ntrel** [comment]
`sizeof(T)` can't be a function because it takes a type parameter. It could be made a generic function, called as `sizeof<T>()`.
Isn't `typeof` going to produce a type rather than a string? If so it can't be a function either.

**Comment by ntrel** [comment]
Because `T.name` is supposed to work, why not have other type properties like `T.sizeof`?

**Comment by spytheman** [comment]
sizeof() has 2 forms - type one `sizeof(int)` and variable one `sizeof(x)`

**Comment by spytheman** [comment]
`embed` was reserved for https://github.com/vlang/rfcs/blob/master/embedding_resources.md

**Comment by spytheman** [comment]
`union` is pretty useful for converting binary data, without explicit casts and manual pointer manipulation

**Comment by ntrel** [comment]
> sizeof() has 2 forms

I know, and the one taking a type can't be a built in function without a generic type parameter.

**Comment by ntrel** [comment]
There's no need to support sizeof on a variable once typeof returns a type, we can just use `typeof(var).sizeof` if sizeof was a type property.

**Comment by spytheman** [comment]
I am merging this, since it is better to have clear docs that reflect the current language even though it may not reflect its ideal form, than an obsolete documentation, that confuses new people (see for example https://discordapp.com/channels/592103645835821068/592294828432424960/754288246237823006 ).

**Pull request merged**